qutip/qip/device/processor.py

```diff
@@ -288,15 +307,19 @@ def get_full_coeffs(self, full_tlist=None):
             full_tlist = self.get_full_tlist()
         coeffs_list = []
         for pulse in self.pulses:
+            if pulse.tlist is None and pulse.coeff is None:
+                coeffs_list.append(np.zeros(len(full_tlist)))
+                continue
             if not isinstance(pulse.coeff, (bool, np.ndarray)):
                 raise ValueError(
                     "get_full_coeffs only works for "
                     "NumPy array or bool coeff.")
             if isinstance(pulse.coeff, bool):
                 if pulse.coeff:
-                    coeffs_list.append(np.ones(full_tlist))
+                    coeffs_list.append(np.ones(len(full_tlist)))
                 else:
-                    coeffs_list.append(np.zeros(full_tlist))
+                    coeffs_list.append(np.zeros(len(full_tlist)))
+                continue
```

Comment:
Good catch. Indeed the logic is similar to `pulse.py`, I think I'll leave this for this PR because (1) In QuTiP 5.0 they will be greatly simplified when the solver allows different `tlist` for different `coeff`; (2）It can be done together with the leakage of private class in `pulse.py`.